### DISCUSSION TITLE
Insights filter using global variable - only if not empty

### DISCUSSION THREAD

**Discussion by meditadvisors** [directus/directus#14259] [Archive]
I am trying to use an input panel that specifies a global variable (Insight module) to filter various panels based this variable
Basically I am trying to implement an Insights search filter but by default (if filter empty) I would like the panels to display all data (i.e. like a "*" wildcard search filter) rather than no data.

I am able to see data when the input field (global variable) is not empty, showing correctly the filtered data. 

However, by default the filter will be empty so how can I enable/configure the filter only if the input/variable is not empty, otherwise show all values as not filtered?

Similar question for a timeline - how can I implement an input date filter FROM - TO that if empty to default as the last month (or whatever is the configured default date interval)

Thanks in advance!

PS: I noticed graphql filter always start with _and, irrespective of the OR/AD operator, for example:
query: "query { query_6b4f6898: test_table (limit: 5, sort: \"id\", filter: **{_and:** [{test_field: {_nempty: true}}, {_or: [{test_field: {_contains: \"test\"}}]}]}) { id test_field} }"
Not sure if this is the cause of this issue since I think an OR iss what is needed for the usecases above...

**Comment by rijkvanzanten** [comment]
I don't think there's currently a way to configure that, as you would have to dynamically apply the filter based on whether or not the value is set 🤔 

Right now, you can do something like:

```
{
  "field": {
    "_eq": "{{ dynamic value }}"
  }
}
```

but that means you end up searching for 

```
{
  "field": {
    "_eq": null
  }
}
```

if the dynamic variable hasn't been set yet. What you're looking for sounds more like:

```
IF dynamic_value THEN { field _eq dynamic value } ELSE { other filter }
```

which is currently not possible

**Reply by meditadvisors** [reply]
Thanks for your quick reply @rijkvanzanten 

Yes, that was the idea, with a slight variation:

`IF dynamic_value THEN { field _eq dynamic value } ELSE {  field _eq * }
`

I was trying something like this, I am not sure if the logic is correct though:

```
{
    "_or": [
        {
            "field": {
                "_eq": "{{ dynamic value }}"
            }
        },
        {
            "field": {
                "_nempty": "true"
            }
        }
    ]
}
```
So return { field _eq dynamic value } OR { field "_nempty": "true" } (that I assumed is a like a "*")

If my logic above does not make sense or cannot be yet implemented then I would appreciate if a feature like this can be considered:
IF dynamic_value is not null THEN { field _eq dynamic value } ELSE {  field _eq * }

**Reply by rijkvanzanten** [reply]
The above attempt is valid, but doesn't achieve the same goal. That says "The field value has to be `{{dynamic value}}` or _anything else_" which basically just means the first half is ignored. What you're looking for is not currently possible

**Reply by meditadvisors** [reply]
Thanks for confirming @rijkvanzanten 

I think this filter functionality would be useful, especially for Insights

If we would like to sponsor this feature, do you have an estimate for implementing this?

**Reply by benhaynes** [reply]
@meditadvisors — you can reach out to us via our website's Live Chat to discuss sponsoring this. ❤️

**Comment by alepietrobon** [comment]
did this ever get picked up? I am facing the same problem. Thx

**Comment by lodgmar** [comment]
Insights won't be really usefull until this will be achieved. I think filters' dropdown should list variables also, not only fields. This way, it'd be easy to filter with a "_or", like this:

Let `field` be a collection's field, and `var` a panel's global variable:
```json
{
   "_or": [
      {
          "var": {
              "_null": true,
          },
          "field": {
              "_eq": "{{ var }}"
          }
      },
  ]
}
```

**Reply by lodgmar** [reply]
Maybe it could use a syntax similar to that of dates, so it would distinguise between fields and variables:

Let `some_variable` be the panel's global variable.

```json
{
   "_or": [
      {
          "var(some_variable)": {
              "_null": true,
          },
          "field": {
              "_eq": "{{ some_variable }}"
          }
      },
  ]
}
```

**Comment by AlexNP23** [comment]
Can this feature be added? As @lodgmar mentioned, Insights is not useful at all without this challenge being resolved. 

Thank you.
